docs/rules.md

```diff
@@ -0,0 +1,121 @@
+
+Discord's `@mentions` system is an extremely useful tool, when used correctly. There are many situations where it's appropriate to use it to get someone's attention, but we discourage the following uses:
+
+* Deliberately mentioning someone when they've asked you to stop doing so
+* Over-mentioning (for example, mentioning someone every single time you address them in a message)
+* Attempting to get the attention of staff for support requests (either in the `#player-support` channel or the `mod-dev` channels)
+
+There's a lot of nuance to consider when deciding whether to mention someone. In general, if someone asks you not to mention them in a specific concept then you should listen to them and change your behaviour.
+
+Additionally, you are able to mention the `@Moderator` role. We ask that you only use this to report problems that are unfolding in public channels and need the attention of a moderator. We will not be upset if the situation ultimately does not need moderation, but it is not appropriate to use this mention to — for example — harass staff members, spam, or draw staff attention to your programming issue.
+
+## Keep all projects legal and appropriate...
+
+> Keep all projects legal and appropriate, and follow licensing requirements — for example, no piracy, malware or troll projects, and all code examples must avoid sexual and discriminatory language
+
+This rule is fairly self-explanitory, but we expect all of your projects to be both legal and appropriate. Although this isn't an exhaustive list, the following is unacceptable:
+
+* Projects that break the Minecraft EULA
+* Projects that enable piracy
+* Projects that contain malicious code, including code that targets specific players
+* Projects that make use of unlicensed (ARR) code from other projects, or otherwise break software licenses from other projects
+* Projects or code examples that contain sexualised or discriminatory language, including in variable names or folder/package structure
+* Projects that promote discrimination or real-life violence
+
+In general, anything that would break another rule if used in casual conversation is not allowed in your code. We expect project maintainers to be professional, respectful and non-discriminatory in their code and interactions.
+
+## Follow the given topics and guidelines for each channel
+
+All of our channels exist for a specific reason. When chatting in any of the channel, your discussions should match the topic the channel was created for. It's pretty normal for conversations to drift off-topic after a while, but the people in that conversation are responsible for moving it to one of the general channels when that happens.
+
+To help you understand what each channel is for, please see our [channel guide](channels.md). You can also look at the description given for each channel on Discord.
+
+## Do not mention names from MCP, Mojang or other proprietary mappings...
```

Comment:
The most common situation we want to avoid is community members coming from other projects using proprietary mappings asking about what X is called in Yarn and posting code snippets, and I believe the rule description could be streamlined for this.

My belief is that a significant amount of modders coming from outside of Fabric are do not fully understand what mappings actually are, not knowing how the names in the code came to be. Thus, it might not be completely obvious that the names they may be taking for granted should not be mentioned in Fabric community spaces. This section does mention that mappings have some relation to names in the code, but does not describe what it means. With a short description of what mappings are, some infractions might be avoided, and it would be easier to explain the rule to community members.

I would also propose an explicit exclusion to this rule to allow posting links to code with proprietary mappings, provided that there is sufficient warning and that the embed (if any) does not reveal anything. AFAIK this has been allowed as long as I have been a member of the Fabric Discord.

Reply:
We will allow links to this and it should be mentioned in the rules.

Reply:
I'll address this in a different order, if that's alright.

---

> I would also propose an explicit exclusion to this rule to allow posting links to code with proprietary mappings, provided that there is sufficient warning and that the embed (if any) does not reveal anything

This is our intention, so I don't see a reason not to make this explicit.

---

> [Mappings]

You raise an interesting point here. I don't really disagree with the sentiment you have here — and we don't want to immediately infract people (either for their first instance, or if it's clear they just didn't know). From that perspective I don't see a rewrite preventing unnecessary infractions, but I think it would be fine for us to clarify the rule a bit.